### PR TITLE
Fix yul codegen bug when using binary negation.

### LINKED ISSUE #11526: [Sol->Yul] ICE because number constant is too small
```
contract C {
  function f() public {
    if (((6) != ~(~(84926290883049832306107864558384249403874903260938453235235091622489261765859)))) {}
  }
}
```

throws

https://github.com/ethereum/solidity/blob/a4f2e591fe55900f86ab72fad37d03af7dfa638e/libsolidity/ast/Types.cpp#L1110

Repro

```
$ solc --experimental-via-ir test.sol
```

### PR DESCRIPTION
**Pull request by Marenz**
I am not sure if this is the way we should fix this, but it is at least
the same way we use in the old codegen...

fixes #11526

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Bugfixes:
  * Code Generator: Fix internal compiler error when calling functions bound to calldata structs and arrays.
  * Code Generator: Fix internal compiler error when passing zero to ``bytes.concat()``.
  * Type Checker: Fix internal error and prevent static calls to unimplemented modifiers.
+ * Yul Code Generator: Fix internal compiler error when using a long literal with binary negation.
```
Comment (chriseth) [inline Changelog.md]:
```suggestion
 * Yul Code Generator: Fix internal compiler error when using a long literal with bitwise negation.
```

Comment (chriseth) [reply Changelog.md]: Actually didn't we say we don't put changelog entries for the ir generator?

File: libsolidity/codegen/ir/IRGeneratorForStatements.cpp [hunk 2]
```diff
@@ -658,12 +658,20 @@ void IRGeneratorForStatements::endVisit(Return const& _return)
...
 	Token const op = _unaryOperation.getOperator();
 
+	if (resultType.category() == Type::Category::RationalNumber)
+	{
+		define(_unaryOperation) << formatNumber(resultType.literalValue(nullptr)) << "\n";
+		return false;
+	}
+
+	_unaryOperation.subExpression().accept(*this);
+
```
Comment (chriseth) [inline libsolidity/codegen/ir/IRGeneratorForStatements.cpp:675]: Need to call setLocation

File: Changelog.md (hunk 1 above)
Comment (Marenz) [reply Changelog.md]: Ah, right. Removing..

**Review by chriseth — APPROVED** [review]

**Pull request merged**